plugins/distance_table.hpp

```diff
@@ -70,24 +70,34 @@ template <class DataFacadeT> class DistanceTablePlugin final : public BasePlugin
     int HandleRequest(const RouteParameters &route_parameters,
                       osrm::json::Object &json_result) override final
     {
-        if (!check_all_coordinates(route_parameters.coordinates))
+        const bool useSameTgtSrc = route_parameters.coordinates.size() ? true : false;
+        if ((useSameTgtSrc && route_parameters.destinations.size()) || (useSameTgtSrc && route_parameters.sources.size()))
+        {
+            return 400;
+        }
+
+        if ((useSameTgtSrc && !check_all_coordinates(route_parameters.coordinates)) ||
+            (!useSameTgtSrc && !check_all_coordinates(route_parameters.destinations, 2) && !check_all_coordinates(route_parameters.sources, 1)))
```

Comment:
`useSameTgtSrc` is unneeded. This can be replaced with:

`if (!check_all_coordinates(route_parameter.coordinates) || !check_all_coordinates(route_parameter.sources) || !check_all_coordinates(route_parameters.destinations)) { return 400; }`